x-pack/plugins/alerting/server/rules_client/methods/bulk_edit.ts

```diff
@@ -302,209 +306,419 @@ async function bulkEditOcc<Params extends RuleTypeParams>(
-              );
-            }
-          }
+async function updateRuleAttributesAndParamsInMemory<Params extends RuleTypeParams>({
+  context,
+  rule,
+  operations,
+  paramsModifier,
+  apiKeysMap,
+  rules,
+  skipped,
+  errors,
+  username,
+}: {
+  context: RulesClientContext;
+  rule: SavedObjectsFindResult<RawRule>;
+  operations: BulkEditOptions<Params>['operations'];
+  paramsModifier: BulkEditOptions<Params>['paramsModifier'];
+  apiKeysMap: ApiKeysMap;
+  rules: Array<SavedObjectsBulkUpdateObject<RawRule>>;
+  skipped: BulkActionSkipResult[];
+  errors: BulkOperationError[];
+  username: string | null;
+}): Promise<void> {
+  try {
+    if (rule.attributes.apiKey) {
+      apiKeysMap.set(rule.id, { oldApiKey: rule.attributes.apiKey });
+    }
 
-          const ruleParams = paramsModifier
-            ? await paramsModifier(attributes.params as Params)
-            : attributes.params;
-
-          // validate rule params
-          const validatedAlertTypeParams = validateRuleTypeParams(
-            ruleParams,
-            ruleType.validate?.params
-          );
-          const validatedMutatedAlertTypeParams = validateMutatedRuleTypeParams(
-            validatedAlertTypeParams,
-            rule.attributes.params,
-            ruleType.validate?.params
-          );
-
-          const {
-            actions: rawAlertActions,
-            references,
-            params: updatedParams,
-          } = await extractReferences(
-            context,
-            ruleType,
-            ruleActions.actions,
-            validatedMutatedAlertTypeParams
-          );
-
-          const shouldUpdateApiKey = attributes.enabled || hasUpdateApiKeyOperation;
-
-          // create API key
-          let createdAPIKey = null;
-          try {
-            createdAPIKey = shouldUpdateApiKey
-              ? await context.createAPIKey(generateAPIKeyName(ruleType.id, attributes.name))
-              : null;
-          } catch (error) {
-            throw Error(`Error updating rule: could not create API key - ${error.message}`);
-          }
+    const ruleType = context.ruleTypeRegistry.get(rule.attributes.alertTypeId);
 
-          const apiKeyAttributes = apiKeyAsAlertAttributes(createdAPIKey, username);
+    await ensureAuthorizationForBulkUpdate(context, operations, rule);
 
-          // collect generated API keys
-          if (apiKeyAttributes.apiKey) {
-            apiKeysMap.set(rule.id, {
-              ...apiKeysMap.get(rule.id),
-              newApiKey: apiKeyAttributes.apiKey,
-            });
-          }
+    const { attributes, ruleActions, hasUpdateApiKeyOperation, isAttributesUpdateSkipped } =
+      await getUpdatedAttributesFromOperations(context, operations, rule, ruleType);
+
+    validateScheduleInterval(
+      context,
+      attributes.schedule.interval as string,
+      ruleType.id,
+      attributes.id as string
```

Comment:
`attributes` has a type RawRule. We do not have field `id` there. But you can use `rule.id` instead.

Reply:
Thanks, fixed!